javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,739 @@
+        return new MapAsJavaMap<>(map);
+    }
+
+    static <K, V> java.util.Map<K, java.util.Collection<V>> asImmutableJava(Multimap<K, V> map) {
+        return new MultimapAsJavaMap<>(map);
+    }
+
+    // -- private view implementations
+    // TODO these should be split out into multiple files
+
+    static abstract class AbstractSeqAsJavaList<T, S extends Seq<T>> extends java.util.AbstractList<T> implements Serializable {
+        private static final long serialVersionUID = 1L;
+        private final ListIteratorFactory<T, S> listIteratorFactory;
+        private Seq<T> delegate;
+
+        AbstractSeqAsJavaList(S delegate, ListIteratorFactory<T, S> listIteratorFactory) {
+            this.delegate = delegate;
+            this.listIteratorFactory = listIteratorFactory;
+        }
+
+        @Override
+        public boolean contains(Object obj) {
+            return narrowed().contains(obj);
+        }
+
+        @Override
+        public boolean containsAll(Collection<?> collection) {
+            return narrowed().containsAll(collection);
+        }
+
+        @Override
+        public T get(int index) {
+            // may throw IndexOutOfBoundsException according to j.u.List.get(int)
+            return delegate.get(index);
+        }
+
+        @Override
+        public int indexOf(Object obj) {
+            return narrowed().indexOf(obj);
+        }
+
+        @Override
+        public boolean isEmpty() {
+            return delegate.isEmpty();
+        }
+
+        @Override
+        public Iterator<T> iterator() {
+            return delegate.iterator();
+        }
+
+        @Override
+        public int lastIndexOf(Object obj) {
+            return narrowed().lastIndexOf(obj);
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public ListIterator<T> listIterator() {
+            return listIteratorFactory.apply((S) delegate, 0);
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public ListIterator<T> listIterator(int index) {
+            return listIteratorFactory.apply((S) delegate, index);
+        }
+
+        @Override
+        public int size() {
+            return delegate.size();
+        }
+
+        @Override
+        public java.util.List<T> subList(int fromIndex, int toIndex) {
+            // may throw IndexOutOfBoundsException according to j.u.List.subList(int, int)
+            return delegate.subSequence(fromIndex, toIndex).asImmutableJavaList();
+        }
+
+        @Override
+        public Object[] toArray() {
+            return delegate.toJavaArray();
+        }
+
+        @Override
+        public <U> U[] toArray(U[] array) {
+            return delegate.toJavaList().toArray(array);
+        }
+
+        public Seq<Object> narrowed() {
+            return Seq.narrow(delegate);
+        }
+
+        // -- Object.*
+
+        @Override
+        public boolean equals(Object o) {
```

Comment:
Can we turn this off in PMD? Or should I just call super.hashCode()? Because that's the implementation I want to use. I considered inheriting super.equals too, but this implementation is probably more efficient.